src/main/java/tsml/classifiers/interval_based/TSF.java

```diff
@@ -339,16 +331,206 @@ public TechnicalInformation getTechnicalInformation() {
         result.setValue(TechnicalInformation.Field.JOURNAL, "Information Sciences");
         result.setValue(TechnicalInformation.Field.VOLUME, "239");
         result.setValue(TechnicalInformation.Field.PAGES, "142-153");
- 
+
         return result;
   }
 
+    /**
+     * buildClassifier wrapper for TimeSeriesInstances
+     *
+     * @param data
+     * @throws Exception
+     */
+    @Override
+    public void buildClassifier(TimeSeriesInstances data) throws Exception {
+        // set classifier capabilities
+        TSCapabilitiesHandler tsCapabilitiesHandler = () -> {
```

Comment:
So the TSF classifier should implement the TSCapabilitiesHandler interface, instead of doing it inline.